src/lua/fiber.c

```diff
@@ -802,20 +802,23 @@ static int
 lbox_fiber_join(struct lua_State *L)
 {
 	struct fiber *fiber = lbox_checkfiber(L, 1);
+	double timeout = TIMEOUT_INFINITY;
 	struct lua_State *child_L = fiber->storage.lua.stack;
 	struct error *e = NULL;
 	int num_ret = 0;
 	int coro_ref = 0;
 
 	if (!(fiber->flags & FIBER_IS_JOINABLE))
 		luaL_error(L, "the fiber is not joinable");
-	fiber_join(fiber);
+	if (lua_isnumber(L, 2))
+		timeout = lua_tonumber(L, 2);
```

Comment:
Please add a check that the second argument is a number, similar to how it's done by luaT_fiber_cond_wait.

Reply:
fixed